kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/fragment/KaalToolbarBindingFragment.kt

```diff
@@ -0,0 +1,180 @@
+     */
+    @DrawableRes
+    open fun getNavIcon(): Int = com.google.android.material.R.drawable.abc_ic_ab_back_material
+
+    /**
+     * Provides a reference to navigation icon description string resource. Default is set to
+     * [com.google.android.material.R.string.abc_action_bar_up_description].
+     *
+     * > **Note:** Overridden method should not call `super()`.
+     */
+    @StringRes
+    open fun getNavIconDescription(): Int = com.google.android.material.R.string.abc_action_bar_up_description
+
+    /**
+     * Creates fragment and sets the information if it has options menu or not.
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(hasMenu)
+    }
+
+    /**
+     * After view is created it will find toolbar (using [findToolbar]), set it up (using [setupToolbar] and set it as
+     * [wrappedToolbar].
+     *
+     * @see findToolbar
+     * @see setupToolbar
+     */
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        wrappedToolbar = (requireActivity() as AppCompatActivity).apply {
+            val toolbar = findToolbar()
+            // if no title is given, the screen / application label would be used
+            if (toolbar.title == null) toolbar.title = ""
+            setSupportActionBar(toolbar)
+            setupToolbar(toolbar)
+        }.supportActionBar!!
+    }
+
+    /**
+     * Inflates the options menu and sets it up by [setupMenu].
+     *
+     * @see setupMenu
+     */
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        if (menuId != null) {
+            inflater.inflate(menuId, menu)
+        }
+        setupMenu(menu)
+    }
+
+    /**
+     * Removes support action bar and destroys [wrappedToolbar] to prevent leaks.
+     */
+    @CallSuper
+    override fun onDestroyView() {
+        (requireActivity() as AppCompatActivity).apply {
+            if (supportActionBar === wrappedToolbar) setSupportActionBar(null)
+        }
+        wrappedToolbar = null
+        super.onDestroyView()
+    }
+
+    /**
+     * Basic toolbar setup. Sets up navigation icon with navigation click and menu item clicks. Navigation click can be
+     * handled using [onNavigationClick]. Menu item click by [onMenuItemClick]. Navigation icon has a drawable and
+     * description which can be modified using [getNavIcon] and [getNavIconDescription].
+     *
+     * @param toolbar Toolbar
+     * @see onNavigationClick
+     * @see onMenuItemClick
+     * @see getNavIcon
+     * @see getNavIconDescription
+     */
+    protected open fun setupToolbar(toolbar: Toolbar) {
+        toolbar.apply {
+            setNavigationOnClickListener {
+                it.hideKeyboard()
+                onNavigationClick(view)
+            }
+            setOnMenuItemClickListener { item -> onMenuItemClick(item) }
```

Comment:
Why don't you use `onOptionsItemSelected` instead of create copy of it?

Reply:
So what should we do about this?

Reply:
We can leave it

Reply:
I will review the API in the future and we will change it if necessary